internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -585,7 +592,7 @@ func (c *Coordinator) Upgrade(ctx context.Context, version string, sourceURI str
 		if errors.Is(err, upgrade.ErrUpgradeSameVersion) {
 			// Set upgrade state to completed so update no longer shows in-progress.
 			det.SetState(details.StateCompleted)
-			return nil
+			return c.upgradeMgr.AckAction(ctx, c.fleetAcker, action)
```

Comment:
What timeout and retry policy does this have? Is their one? 

If Fleet Server is unavailable for an hour and all acks are failing what is the behavior of this call?

Reply:
i went with simple case with no retries for readability and that the probability with immediate ack is low. 
it can still happen though. i refactored code to pass things around from the top but it don't like it that much, it makes this a mess